src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -394,7 +395,7 @@ function initializeNetworkFiltering(frb: FunctionsRuntimeBundle): void {
         .filter((v) => v);
       const href = (hrefs.length && hrefs[0]) || "";
 
-      if (href && !history[href]) {
+      if (href && !history[href] && !href.startsWith("http://localhost")) {
```

Comment:
Maybe use the URL module? https://nodejs.org/docs/latest-v8.x/api/url.html#url_constructor_new_url_input_base

You can get the `.hostname` to check that it's going to `localhost`

Reply:
This one I am going to leave because I am actually not sure `href` will always parse nicely and I don't want to risk an error down here.